modules/twinklefluff.js

```diff
@@ -80,101 +80,97 @@ Twinkle.fluff = {
+			if( document.getElementById('differences-nextlink') ) {
+				// Not latest revision
+				var revertToRevisionN = document.createElement('div');
+				revertToRevisionN.setAttribute( 'id', 'tw-revert-to-nrevision' );
+				revertToRevisionN.style.fontWeight = 'bold';
+
+				var revertToRevisionNLink = buildLink('SaddleBrown', 'restore this version');
+				revertToRevisionNLink.href = "#";
+				$(revertToRevisionNLink).click(function(){
+					Twinkle.fluff.revertToRevision(mw.config.get('wgDiffNewId').toString());
+				});
+				revertToRevisionN.appendChild(revertToRevisionNLink);
+
+				ntitle.insertBefore( revertToRevisionN, ntitle.firstChild );
+			} else if( Twinkle.getPref('showRollbackLinks').indexOf('diff') !== -1 && mw.config.get('wgDiffOldId') && (mw.config.get('wgDiffOldId') !== mw.config.get('wgDiffNewId') || document.getElementById('differences-prevlink'))) {
+				var vandal = $("#mw-diff-ntitle2").find("a").first().text();
+
+				var revertNode = document.createElement('div');
+				revertNode.setAttribute( 'id', 'tw-revert' );
+
+				var agfNode = document.createElement('strong');
+				var vandNode = document.createElement('strong');
+				var normNode = document.createElement('strong');
+
+				var agfLink = buildLink('DarkOliveGreen', 'rollback (AGF)');
+				var vandLink = buildLink('Red', 'rollback (VANDAL)');
+				var normLink = buildLink('SteelBlue', 'rollback');
+
+				agfLink.href = "#";
+				vandLink.href = "#";
+				normLink.href = "#";
+				$(agfLink).click(function(){
+					Twinkle.fluff.revert('agf', vandal);
+				});
+				$(vandLink).click(function(){
+					Twinkle.fluff.revert('vand', vandal);
+				});
+				$(normLink).click(function(){
+					Twinkle.fluff.revert('norm', vandal);
+				});
+
+				agfNode.appendChild(agfLink);
+				vandNode.appendChild(vandLink);
+				normNode.appendChild(normLink);
+
+				revertNode.appendChild( agfNode );
+				revertNode.appendChild( document.createTextNode(' || ') );
+				revertNode.appendChild( normNode );
+				revertNode.appendChild( document.createTextNode(' || ') );
+				revertNode.appendChild( vandNode );
+
+				ntitle.insertBefore( revertNode, ntitle.firstChild );
 			}
-			otitle = otitle1.parentNode;
-			ntitle = ntitle1.parentNode;
-		} catch( e ) {
-			// no old, nor new title, nothing to do really, return;
-			return;
 		}
+	},
 
-		// Lets first add a [restore this revision] link
+	oldid: function() { // Add a [restore this revision] link on old revisions
 		var revertToRevision = document.createElement('div');
 		revertToRevision.setAttribute( 'id', 'tw-revert-to-orevision' );
 		revertToRevision.style.fontWeight = 'bold';
 
 		var revertToRevisionLink = buildLink('SaddleBrown', 'restore this version');
-
 		revertToRevisionLink.href = "#";
 		$(revertToRevisionLink).click(function(){
-			Twinkle.fluff.revertToRevision(mw.config.get('wgDiffOldId').toString());
+			Twinkle.fluff.revertToRevision(mw.config.get('wgRevisionId').toString());
```

Comment:
Fair warning, because this tripped me up for a while, but the regular split diff view makes it look like `wgDiffOldId` is being changed to `wgRevisionId`, but it's not!  `git diff` does poorly with duplicated sections, so the stuff on the left (diff) is actually above this section (oldid).  Viewing without whitespace should make it clearer.